javaext/com.microsoft.java.lombok/src/main/java/com/microsoft/java/lombok/ConstructorHandler.java

```diff
@@ -0,0 +1,100 @@
+import org.eclipse.jdt.core.dom.Modifier;
+import org.eclipse.jdt.core.dom.ASTNode;
+import org.eclipse.jdt.core.dom.CompilationUnit;
+import org.eclipse.jdt.core.manipulation.CoreASTProvider;
+import org.eclipse.jdt.internal.corext.dom.ASTNodes;
+import org.eclipse.jdt.ls.core.internal.JavaLanguageServerPlugin;
+import org.eclipse.jdt.ls.core.internal.preferences.Preferences;
+import org.eclipse.jdt.internal.corext.codemanipulation.CodeGenerationSettings;
+import org.eclipse.jdt.ls.core.internal.handlers.GenerateConstructorsHandler;
+import org.eclipse.jdt.ls.core.internal.handlers.GenerateConstructorsHandler.CheckConstructorsResponse;
+import org.eclipse.jdt.ls.core.internal.handlers.JdtDomModels.LspMethodBinding;
+import org.eclipse.jdt.ls.core.internal.handlers.JdtDomModels.LspVariableBinding;
+import org.eclipse.jdt.ls.core.internal.text.correction.SourceAssistProcessor;
+import org.eclipse.jdt.core.dom.rewrite.ListRewrite;
+import org.eclipse.lsp4j.CodeActionParams;
+import org.eclipse.text.edits.TextEdit;
+
+import com.microsoft.java.lombok.ConstructorHandler;
+
+public class ConstructorHandler {
+    public static void removeMethods(IType type, ListRewrite rewriter, ConstructorKind kind, IProgressMonitor monitor) {
+        try {
+            CompilationUnit astRoot = CoreASTProvider.getInstance().getAST(type.getCompilationUnit(),
+                    CoreASTProvider.WAIT_YES, monitor);
+            if (astRoot == null) {
+                return;
+            }
+            ITypeBinding typeBinding = ASTNodes.getTypeBinding(astRoot, type);
+            if (typeBinding == null) {
+                return;
+            }
+            
+            long nonStaticFieldLength = Arrays.stream(typeBinding.getDeclaredFields()).filter(field -> !Modifier.isStatic(field.getModifiers())).count();
+
+            for (IMethodBinding item : typeBinding.getDeclaredMethods()) {
+                if (item.isDefaultConstructor()) {
+                    continue;
+                }
+                int parametersNum = item.getParameterTypes().length;
+                if (item.getName().equals(typeBinding.getName())) {
+                    if (kind == ConstructorKind.NO_ARG && parametersNum == 0) {
+                        ASTNode node = astRoot.findDeclaringNode(item);
+                        rewriter.replace(node, null, null);
+                        break;
+                    } else if (kind == ConstructorKind.ALL_ARGS && parametersNum == nonStaticFieldLength) {
+                        ASTNode node = astRoot.findDeclaringNode(item);
+                        rewriter.replace(node, null, null);
+                        break;
+                    }
+                }
+            }
+
+        } catch (Exception e) {
+            JavaLanguageServerPlugin.logException("Remove Lombok method", e);
+        }
+        return;
+    }
+
+    public static TextEdit generateConstructor(CodeActionParams params, IProgressMonitor monitor,
```

Comment:
When generating methods, you should reference the existing lombok annotation as well. For example, I have added two annotations (@Data, @NoArgsConstructor) above a class, delomboking @NoArgsConstructor should not generate constructors.

Reply:
A more complex case is to delombok and lombok in one code action.

https://user-images.githubusercontent.com/14052197/191197452-6e12ae72-9200-4827-94c5-78236c7b4fc1.mp4